src/v/cloud_topics/level_one/compaction/worker.cc

```diff
@@ -239,9 +253,42 @@ ss::future<> compaction_worker::resume_worker() {
         co_return;
     }
 
+    // If `_work_fut` still has a value, it is because the request to
+    // `resume_worker()` was received before `_work_fut` was stopped by
+    // `pause_worker()`. Wait for it to be cleared before launching a new
+    // background fiber.
+    if (_work_fut.has_value()) {
+        vlog(
+          compaction_log.info,
+          "Waiting for backgrounded compaction process to finish before "
+          "resuming worker");
+    }
+
+    static constexpr std::chrono::seconds work_fut_timeout(60);
+    while (_work_fut.has_value()) {
+        auto wait_fut = co_await ss::coroutine::as_future(
+          _work_fut_cv.wait(work_fut_timeout));
+        if (wait_fut.failed()) {
```

Comment:
yeh part of the complexity is that _work_fut is being modified concurrently here and in clear_work_fut (and other places?)